addons/knowledge/models/knowledge_article.py

```diff
@@ -1461,6 +1497,36 @@ def _send_invite_mail(self, partners):
                 subject=subject,
             )
 
+    # ------------------------------------------------------------
+    # Article Items
+    # ------------------------------------------------------------
+
+    def _get_article_item_move_values(self, target_parent_article):
+        """ This method is called during write override and handles the logic
+        behind article items move. Under certain conditions, an article, may it
+        be normal or an item, will change its type to respect the constraints:
+            - Article items must have a parent
+            - A parent can only contain on type of child articles.
+
+        Therefore, an article must be(come) normal:
+            - When an article becomes root (as article items must have a parent)
+            - When putting an article under a parent with 'normal' articles
+        And an article must be(come) an item:
+            - When putting an article under parent with article items
+
+        :param target_parent_article: Model<knowledge.article> new parent of the
+        current self recordset. May be empty if articles in self are moved to
+        root.
+        """
+        values = {}
+        has_article = any(not article.is_article_item for article in self)
+        has_item = any(article.is_article_item for article in self)
+        if has_item and not target_parent_article or target_parent_article.article_child_ids:
+            values['is_article_item'] = False
+        elif has_article and target_parent_article.item_child_ids:
+            values['is_article_item'] = True
```

Comment:
This computation is strange. Won't it crash anyway if you have both article and items ?